src/main/java/org/example/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+import org.example.bookstore.repository.orderitem.OrderItemRepository;
+import org.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import org.example.bookstore.repository.user.UserRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemsMapper orderItemsMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createNewOrder(OrderRequestDto requestDto,
+                                           Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("User with id: " + userId
+                        + "does not exist"));
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findShoppingCartByUserId(user.getId()).orElseThrow(()
+                        -> new EntityNotFoundException("Shopping cart does not exist"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        if (cartItems == null || cartItems.isEmpty()) {
+            throw new EmptyCartException("Your shopping cart is empty ");
+        }
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setShippingAddress(requestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = cartItems.stream().map(cartItem -> {
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItem.setPrice(cartItem.getBook().getPrice().multiply(
+                    BigDecimal.valueOf(cartItem.getQuantity())));
+
+            return orderItem;
+        }).collect(Collectors.toSet());
+        order.setOrderItems(orderItems);
+        order.setTotal(calculateTotal(orderItems));
+
+        Order savedOrder = orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        cartItemRepository.deleteAllByShoppingCartId(shoppingCart.getId());
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<OrderResponseDto> getAllUserOrders(Long userId,Pageable pageable) {
+        User user = userRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException(
+                "can`t find user " + userId));
```

Comment:
This logic should belong to user service, or you can accept user as parameter in method to avoid DB requiest

Reply:
+